kyaml/kio/byteio_reader.go

```diff
@@ -17,8 +17,14 @@ import (
 )
 
 const (
-	ResourceListKind       = "ResourceList"
-	ResourceListAPIVersion = "config.kubernetes.io/v1"
+	ResourceListKind = "ResourceList"
+
+	ResourceListAPIVersionV1       = "config.kubernetes.io/v1"
+	ResourceListAPIVersionV1Beta1  = "config.kubernetes.io/v1beta1"
+	ResourceListAPIVersionV2Alpha1 = "config.kubernetes.io/v2alpha1"
```

Comment:
Do we want to deprecate the old versions?

Reply:
I'd be OK with deprecating `v1beta1` but I don't think I have a say on v1 :/

Reply:
+1 deprecating `v1beta1`

Reply:
I marked it to be explicitly deprecated (hopefully won't trigger any linter in this PR lol)